bugbug/models/worksforme.py

```diff
@@ -0,0 +1,118 @@
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, utils
+from bugbug.model import BugModel
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class WorksForMeModel(BugModel):
+    def __init__(self, lemmatization=False):
+        BugModel.__init__(self, lemmatization)
+
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.HasSTR(),
+            bug_features.HasRegressionRange(),
+            bug_features.Status(),
+            bug_features.Severity(),
+            bug_features.Priority(),
+            bug_features.HasURL(),
+            bug_features.Whiteboard(),
+            bug_features.Product(),
+            bug_features.Component(),
+            bug_features.Keywords(),
+            bug_features.TimeToClose(),
+            bug_features.HasAttachment(),
+            bug_features.CommentCount(),
+            bug_features.CommentLength(),
+            bug_features.NumWordsComments(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+            feature_cleanup.hex(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(
+                        feature_extractors,
+                        cleanup_functions,
+                        rollback=True,
+                        rollback_when=self.rollback,
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = ImblearnPipeline(
+            [
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.0001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", BorderlineSMOTE(random_state=0)),
+                (
+                    "estimator",
+                    xgboost.XGBClassifier(n_jobs=utils.get_physical_cpu_count()),
+                ),
+            ]
+        )
+
+    def rollback(self, change):
+        return change["field_name"] == "cf_last_resolved" and change["added"]
+
+    def get_labels(self):
+        classes = {}
+
+        for bug in bugzilla.get_bugs():
+            bug_id = int(bug["id"])
+
+            if not bug["resolution"] and not bug["cf_last_resolved"]:
+                continue
```

Comment:
What type of cases are excluded here?

Reply:
> What type of cases are excluded here?

It excludes open bugs

Reply:
I would use `or` instead of `and`, so that will exclude any bug that was previously closed and reopened.